### ISSUE TITLE
anyone using redux-form with this ?

### ISSUE THREAD

**Issue by rcholic** [mxstbr/react-boilerplate#567]
I am trying to integrate `redux-form` with this boilerplate, but so far, no luck getting it to work. I keep getting this error:
```
createHigherOrderComponent.js 
Uncaught (in promise) Error: You need to mount the redux-form reducer at "form"(…)
```

I follow this example [here](http://redux-form.com/5.2.5/#/examples/synchronous-validation?_k=i8u7d2) by creating a dumb component that contains the form, which is imported into a container. 

Or is there any form plugin/module you guys recommend using with this boilerplate? Thanks

**Comment by gihrig** [comment]
@rcholic Here's a pretty complete [example](https://react-redux.herokuapp.com/survey), with [container source](https://github.com/erikras/react-redux-universal-hot-example/blob/master/src/containers/Survey/Survey.js) and [component source](https://github.com/erikras/react-redux-universal-hot-example/tree/master/src/components/SurveyForm). The example shows `dirty`, `pristine`, `valid` and `invalid` properties.

I expect to integrate redux-form into my app, based on this boilerplate, and this is the example I plan to follow. Please let us know how you get on with it. Hope this helps.

**Comment by rcholic** [comment]
@gihrig thanks for your link to the example, but I was well aware of that example. I don't this boilerplate supports the decorator syntax as used in the example, so I wrote the following in replacing the decorator:

```
const regForm = reduxForm({
    form: 'registrationForm',
    fields: fields,
    validate: validate
})
```

To export the dumb form component, I did the following:
```
export default connect(()=>({}))(regForm(UserRegistrationForm));
```
or
```
export default regForm(UserRegistrationForm);
```

Neither works in my case, and I keep getting the error as in the first post.

I make sure I have the following in my `reducer.js` file:
```
import { reducer as formReducer } from 'redux-form'; // redux-form reducer

return combineReducers({
    route: routeReducer,
    global: globalReducer,
    form: formReducer, // mounted at 'form'
    questionList: questionListReducer,
    questionView: questionViewReducer,
    commentList: commentListReducer,
    answerList: answerListReducer,
    headerNavBar: headerNavBarReducer,
    login: loginReducer,
    userRegistration: userRegistrationReducer,
    ...asyncReducers,
  });
```

It does not work for me, but please let me know if you can make `redux-form` work with this boilerplate. Thanks again!

**Comment by igorbarbashin** [comment]
redux-form 5 doesn't work with immutableJS state out of the box.
So instead of `form: formReducer` you need something like this:
`form: (state = Immutable.fromJS({}), action) => Immutable.fromJS(formReducer(state.toJS(), action)),`

You'll also need to import immutable at the top of the file:
`import Immutable from 'immutable';`

And in you reduxForm decorator you need additional argument `getFormState`:

```
export default reduxForm({
  form: 'MyForm',
  fields,
  getFormState: (state, reduxMountPoint) => state.get(reduxMountPoint).toJS(),
})(MyForm);
```

**Comment by tobyl** [comment]
Hi @gihrig - you've posted many great tips in response to issues with redux-form, many thanks. Just wondering there's any more recent code samples floating around - specifically for v6.0.0-rc.3 of redux-form?

**Comment by zanedev** [comment]
We have successfully integrated v6 rc3 and there isn't that much required other than how you map your fields to your selectors, they're nested differently under another child "input" object so it does require some refactoring.  The new redux-form will have immutable support out of the box and you won't need to to the getFormState: toJS but we are finding out there are still a few quirks with the immutable support so we haven't switched completely to it and still using the toJS method.

**Comment by luandro** [comment]
Isn't redux-form a standard in dealing with forms in a React/Redux app? Shouldn't we consider supporting it out of the box, like we're doing with helmet and i18n?

**Comment by mxstbr** [comment]
> Isn't redux-form a standard in dealing with forms in a React/Redux app?

I've never used it, and only heard of it because of this issue 😁 Sample size = 1, I know, but it's definitely not a standard for me!

**Comment by luandro** [comment]
@mxstbr Just out of curiosity, how do you usually deal with forms? Using React state?

**Comment by mxstbr** [comment]
Yeah, I've never had a really form-intensive app. I either put them in the local state or it was a login form where having a `username` field in the global state is beneficial anyway :+1:

**Comment by tobyl** [comment]
@zanedev I appreciate the feedback, very helpful, thank you! For the record, I don't necessarily believe redux-form should be part of the boilerplate, but it does seems to be a well accepted solution to handling forms in react/redux.

**Comment by BugsBunny338** [comment]
Since redux-form@v6 simply http://redux-form.com/6.0.2/docs/faq/ImmutableJs.md/.

**Comment by eduardoinnorway** [comment]
The above example does not work out of the box with React-Boilerplate, who ever has it working I suggest  you make a working example please. And you should support Redux-Form out of the box.

**Comment by tomByrer** [comment]
It seem [formik](https://github.com/jaredpalmer/formik/issues/84) is more performant.
